### ISSUE TITLE
Redesigned homepage: Email address and search field input including form action/testing

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-cms#11389]
## Issue description
We received accessibility feedback during Midpoint review (Reference Midpoint Review - Accessibility Feedback - Sitewide Homepage, VA.gov homepage redesign [#48585](https://github.com/department-of-veterans-affairs/va.gov-team/issues/48585)).

In the current implementation, aria-labels provide an accessible name for screen readers. @briandeconinck suggested using the <label> element to add a visible label for the Email address and search field input fields. **_This was listed as "should" feedback and should be addressed before launch._**

The redesigned homepage uses the design system va-search-input component so we'll likely need to wait for the component to be updated. @briandeconinck and @laflannery to follow-up with DST to learn any other options. 

### Form action
Form action = `action="https://public.govdelivery.com/accounts/USVACHOOSE/subscribers/qualify"`
Based on implementation from https://www.choose.va.gov/. 

## Background and rationale
For sighted voice command software users (eg. Dragon Naturally Speaking), moving focus to a form control is accomplished by speaking a command like "Click [name of control]." That requires the user knowing what the name of the control is --- which requires having a label they can see. In this context, both the search field and the email address field are pretty guessable, and most users will probably get enough of a match that their software can get them to the right place. But it's a potential point of friction that can be removed easily with a visible label.

## Pending
- [ ] Mobile design mockup from Ryan

## Screenshots

Reference: [Sketch Round 3](https://www.sketch.com/s/3aa40506-4be2-46cc-876b-93f1a9f3a857/p/EAD305BC-3082-4F81-ADEC-C2486F7B6BEF/canvas)

<details> 
<summary> Desktop and Mobile designs </summary>

![image.png](https://images.zenhubusercontent.com/5d8a47393ac3440001d1a2ed/710ff412-a213-4767-9bf6-26ce0ca773ed)

</details>

## Tasks
- [x] Update implementation consistent with [Sketch Round 3](https://www.sketch.com/s/3aa40506-4be2-46cc-876b-93f1a9f3a857/p/EAD305BC-3082-4F81-ADEC-C2486F7B6BEF/canvas)

## Acceptance criteria
Veterans and beneficiaries viewing the redesigned homepage can signup for email updates using a component at the bottom of the page which 
- [x] uses a stacked design in desktop and mobile
- [x] has intro text ""Sign up to get the latest VA updates"
- [x] has the label "Email address" above the field
- [x] has actionable "Sign up" button below the field. 

- [ ] verify successful submission _with stakeholders_ (TBD how from @mmiddaugh ) 
- [ ] approval/sign off from Laura Flannery

**Comment by wesrowe** [comment]
@mmiddaugh, I assigned back to you because this needs Ryan T to provide a label to engineering.

**Comment by mmiddaugh** [comment]
Discussed with Ryan today - design input to be provided by 11/11/2022

**Comment by jilladams** [comment]
Closing #11511 as a dupe of this: 
* The Design System doesn't have an explicit email field
* The [Text Input](https://design.va.gov/storybook/?path=/docs/components-va-text-input--default) does accept 'type' and 'inputmode' attributes, which can be set to 'email' and provides with own validation UI. 
* Looking at the [forthcoming design iteration](https://www.sketch.com/s/3aa40506-4be2-46cc-876b-93f1a9f3a857/p/EAD305BC-3082-4F81-ADEC-C2486F7B6BEF/canvas), it seems that may have been taken into account and the UI now matches the standard form input as the Design System provides.
 * Assuming the form submission works as intended, there's rather low LOE on implementing this from the front end perspective.

**Comment by mmiddaugh** [comment]
Per @davidconlon, this form flows into the subscribe flow on GovDelivery, which is where the error states will occur. Behavior can be replicated on choose.va.gov.

**Comment by jilladams** [comment]
Carmel Hinkle (carmel.hinkle@granicus.com) & Alec Manriquez: non-technical business contacts for Granicus. Granicus is parent company that owns / runs govdelivery. 

* Our form will submit emails to govdelivery
* Currently, VA has an account you can get to via "Email Updates" link in the VA.gov footer. We are not confirmed whether we will use the USVA or USVACHOOSE accounts for submission. (Org/account is included in the govdelivery paths.)
  * Form action will be something like action="https://public.govdelivery.com/accounts/USVA/subscribers/qualify", example on choose.va.gov 
* Testing: we will need a login to govdelivery in order to verify successful submission. However: 
  * All error messaging / validation happens on the govdelivery side / isn't ours to manage.
  * Successfully delivering users to the govdelivery URL = success for us. (To be confirmed)

**Action items:**
* @mmiddaugh to confirm whether we need to use Choose VA or USVA version of the govdelivery account for our form action

**Comment by jilladams** [comment]
My takeaway here @mmiddaugh @wesrowe is that Bryan is actually blocked and we can't do much here til we confirm those open action items?

**Comment by wesrowe** [comment]
I concur, but he may not (or may) be blocked for long – the action item for next step is that Alec is opening a ticket with the Tech Support group to assist us. We didn't get an ETA but I would expect it pretty quickly. May have to re-estimate the ticket though.

**Comment by apisandipas** [comment]
@jilladams @wesrowe @mmiddaugh 

I've implemented the design changes and the form, copying the implementation from https://choose.va.gov in this PR [here](https://github.com/department-of-veterans-affairs/content-build/pull/1386)

However, the form lands on an error page, saying that the email provided was empty. I've verified the field names are correct and match the implementation on the Choose VA website to the best of my knowledge.

I've consulted the [documentation provided](https://support.granicus.com/s/article-group/a8C4w000000XcOdEAK/govdelivery-api?p=govDelivery%20Suite&p=) by Granicus, but it appears to detail a JSON API rather than a form endpoint for us to “post to” like the Choose VA website appears to do. The documentation appears rather newly updated (early October 2022) so perhaps this is a new preferred way for us to interact with that endpoint.

If so, we could implement an asynchronous JavaScript form submission of the email according to their documentation. This would likely involve doing the error handling on our end of things, though, which is not what was specified originally.  FWIW, the option of implementing the form with asynchronous submission to an API endpoint would be much greater and likely involve the creation of a new React-based widget to handle form state, error messaging etc.

Either way, we will likely need to reach out to the Granicus folks again to see if there's a way to make the current implementation work.
